### PR TITLE
Pia update

### PR DESCRIPTION
**Pull request by insane-04**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { NgModule } from '@angular/core';
 import { Routes, RouterModule, PreloadAllModules } from '@angular/router';
 import { DocumentSearchModule } from './modules/document-search/document-search.module';
 
-const routes: Routes = [
+export const routes: Routes = [
   {
     path: '',
     redirectTo: '/search',
```
Comment (SchettlerKoehler) [inline src/app/app-routing.module.ts:8]: missing addInitializeModuleGuard

File: src/app/modules/document-detail/document-create/document-attachment/document-attachment.component.ts [hunk 2]
```diff
@@ -50,10 +55,12 @@ export class DocumentAttachmentComponent implements OnInit {
   constructor(
     private readonly translateService: TranslateService,
     private readonly supportedMimeTypeV1Service: SupportedMimeTypeControllerV1APIService,
-    private readonly dataSharingService: DataSharingService
+    private readonly dataSharingService: DataSharingService,
+    private userService: UserService
   ) {}
 
   ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-create/document-attachment/document-attachment.component.ts:63]: Not needed if you use addInitializeModuleGuard. It is not safe in all cases to do it here!

File: src/app/utils.ts [hunk 3]
```diff
@@ -190,3 +195,19 @@ export function noSpecialCharacters(
   const pattern = /[\\/:*?<>|"]/;
   if (pattern.test(control.value)) return { hasSpecialChars: true };
 }
+
+/**
+ * @param userService - The `UserService` instance that provides the user's language preference observable (`lang$`).
+ * @returns A promise that resolves to the best matching language as a string, which will be either 'en', 'de', or the default 'en'.
+ */
+export function getBestMatchLanguage(userService: UserService): string {
```
Comment (SchettlerKoehler) [inline src/app/utils.ts:203]: This function is not working in all cases. Please never ever use subscribe to get a value out of an observable!!! This is leading to runtime errors which are very hard to find and very hard to reproduce

File: src/app/modules/document-detail/document-create/document-create.component.spec.ts [hunk 4]
```diff
@@ -33,14 +45,18 @@ describe('DocumentCreateComponent', () => {
         HttpClientTestingModule,
         RouterTestingModule,
         ReactiveFormsModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-create/document-create.component.spec.ts:48]: You can and should use the TranslateTestingModule

File: src/app/modules/document-detail/document-create/document-create.component.ts [hunk 5]
```diff
@@ -55,14 +59,16 @@ export class DocumentCreateComponent implements OnInit {
...
+    private readonly portalMessageService: PortalMessageService,
     private readonly attachmentUploadService: AttachmentUploadService,
+    private userService: UserService,
     private readonly router: Router,
     private readonly activeRoute: ActivatedRoute,
     private readonly formBuilder: UntypedFormBuilder
   ) {}
 
   ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-create/document-create.component.ts:71]: Not needed. See my comment above

File: src/app/modules/document-detail/document-detail-routing.module.ts [hunk 6]
```diff
@@ -6,23 +6,19 @@ import { Routes, RouterModule } from '@angular/router';
 import { DocumentCreateComponent } from './document-create/document-create.component';
 import { DocumentEditComponent } from './document-edit/document-edit.component';
 import { DocumentQuickUploadComponent } from './document-quick-upload/document-quick-upload.component';
-import { CanActivateGuard } from 'src/app/shared/can-active-guard.service';
 
 const routes: Routes = [
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-detail-routing.module.ts:10]: Use addInitializeModuleGuard

File: src/app/modules/document-detail/document-edit/document-edit-attachment/document-edit-attachment.component.spec.ts [hunk 7]
```diff
@@ -27,15 +36,25 @@ describe('DocumentEditAttachmentComponent', () => {
   beforeEach(async () => {
     const translateSpy = jasmine.createSpyObj('TranslateService', ['get']);
     await TestBed.configureTestingModule({
-      imports: [HttpClientTestingModule, FormsModule, ReactiveFormsModule],
+      imports: [
+        HttpClientTestingModule,
+        FormsModule,
+        ReactiveFormsModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-edit/document-edit-attachment/document-edit-attachment.component.spec.ts:43]: Use TranslateTestingModule

File: src/app/modules/document-detail/document-edit/document-edit-attachment/document-edit-attachment.component.ts [hunk 8]
```diff
@@ -56,10 +61,12 @@ export class DocumentEditAttachmentComponent implements OnInit {
     private readonly translateService: TranslateService,
     private readonly supportedMimeTypeV1Service: SupportedMimeTypeControllerV1APIService,
     private readonly attachmentUploadService: AttachmentUploadService,
-    private readonly dataSharingService: DataSharingService
+    private readonly dataSharingService: DataSharingService,
+    private userService: UserService
   ) {}
 
   ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-edit/document-edit-attachment/document-edit-attachment.component.ts:69]: Not needed. See my comment above

File: src/app/modules/document-detail/document-edit/document-edit.component.spec.ts [hunk 9]
```diff
@@ -55,13 +66,19 @@ describe('DocumentEditComponent', () => {
         BrowserModule,
         ReactiveFormsModule,
         FormsModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-edit/document-edit.component.spec.ts:69]: Use TranslateTestingModule

File: src/app/modules/document-detail/document-edit/document-edit.component.spec.ts [hunk 10]
```diff
@@ -195,10 +213,10 @@ describe('DocumentEditComponent', () => {
     component.attachmentFormValidation(formFieldArray);
     expect(
       formFieldArray.has('DOCUMENT_MENU.DOCUMENT_EDIT.ATTACHMENT_NAME_MISSING')
-    ).toBe(false);
+    ).toBe(true);
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-edit/document-edit.component.spec.ts:216]: Why has this changed?

File: src/app/modules/document-detail/document-edit/document-edit.component.ts [hunk 11]
```diff
@@ -35,7 +37,11 @@ import {
   UpdateDocumentRequestParams,
 } from 'src/app/generated';
 import { DocumentTabStateService } from 'src/app/shared/document-tab-state.service';
-import { convertToCSV, noSpecialCharacters } from 'src/app/utils';
+import {
+  convertToCSV,
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-edit/document-edit.component.ts:41]: Use the ExportService here

File: src/app/modules/document-detail/document-edit/document-edit.component.ts [hunk 12]
```diff
@@ -78,18 +84,20 @@ export class DocumentEditComponent implements OnInit {
...
     private readonly attachmentUploadService: AttachmentUploadService,
     private readonly router: Router,
     public documentTabStateService: DocumentTabStateService,
     private readonly dataSharingService: DataSharingService,
+    private userService: UserService,
     private readonly datepipe: DatePipe
   ) {}
 
   public ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-edit/document-edit.component.ts:100]: Not needed. See my comment above

File: src/app/modules/document-detail/document-edit/document-edit.component.ts [hunk 13]
```diff
@@ -253,7 +259,7 @@ export class DocumentEditComponent implements OnInit {
   private setHeaderButtons(document: DocumentDetailDTO) {
     this.headerActions = [
       {
-        label: this.translatedData['GENERAL.CLOSE'],
+        label: this.translateService?.instant('GENERAL.CLOSE'),
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-edit/document-edit.component.ts:262]: Do not use translateService.instant(). This is not safe. Especially with the way you are loading the translations. headerActions should be an observable which depends on translateService.get()

File: src/app/modules/document-detail/document-quick-upload/document-quick-upload-form/document-quick-upload-form.component.spec.ts [hunk 14]
```diff
@@ -33,11 +42,15 @@ describe('DocumentQuickUploadFormComponent', () => {
         BrowserModule,
         ReactiveFormsModule,
         FormsModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-quick-upload/document-quick-upload-form/document-quick-upload-form.component.spec.ts:45]: Use TranslateTestingModule

File: src/app/modules/document-detail/document-quick-upload/document-quick-upload-form/document-quick-upload-form.component.ts [hunk 15]
```diff
@@ -60,10 +65,12 @@ export class DocumentQuickUploadFormComponent implements OnInit {
     private readonly documentTypeV1Service: DocumentTypeControllerV1APIService,
     private readonly supportedMimeTypeV1Service: SupportedMimeTypeControllerV1APIService,
     private readonly translateService: TranslateService,
+    private userService: UserService,
     private readonly dataSharingService: DataSharingService
   ) {}
 
   ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-quick-upload/document-quick-upload-form/document-quick-upload-form.component.ts:73]: Not needed. See my comment above

File: src/app/modules/document-detail/document-quick-upload/document-quick-upload.component.spec.ts [hunk 16]
```diff
@@ -38,13 +45,19 @@ describe('DocumentQuickUploadComponent', () => {
         BrowserModule,
         ReactiveFormsModule,
         FormsModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-quick-upload/document-quick-upload.component.spec.ts:48]: Use TranslateTestingModule

File: src/app/modules/document-detail/document-quick-upload/document-quick-upload.component.ts [hunk 17]
```diff
@@ -76,16 +81,18 @@ export class DocumentQuickUploadComponent implements OnInit {
...
     private readonly router: Router,
     private readonly activeRoute: ActivatedRoute,
     private readonly breadcrumbService: BreadcrumbService,
-    private readonly translateService: TranslateService
+    private readonly translateService: TranslateService,
+    private userService: UserService
   ) {}
 
   ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-detail/document-quick-upload/document-quick-upload.component.ts:95]: Not needed. See my comment above

File: src/app/modules/document-more/document-bulk-changes/document-bulk-changes.component.spec.ts [hunk 18]
```diff
@@ -40,10 +49,20 @@ describe('DocumentBulkChangesComponent', () => {
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       declarations: [DocumentBulkChangesComponent, TranslatePipeMock],
-      imports: [ReactiveFormsModule, HttpClientTestingModule],
+      imports: [
+        ReactiveFormsModule,
+        HttpClientTestingModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-more/document-bulk-changes/document-bulk-changes.component.spec.ts:55]: Use TranslateTestingModule

File: src/app/modules/document-more/document-bulk-changes/document-bulk-changes.component.ts [hunk 19]
```diff
@@ -62,17 +68,19 @@ export class DocumentBulkChangesComponent implements OnInit, OnDestroy {
...
     private readonly activeRoute: ActivatedRoute,
-    private readonly messageService: MessageService,
+    private readonly portalMessageService: PortalMessageService,
     private readonly userDetailsService: UserDetailsService,
     private readonly dataSharingService: DataSharingService,
     private readonly documentTypeV1Service: DocumentTypeControllerV1APIService
   ) {}
 
   ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-more/document-bulk-changes/document-bulk-changes.component.ts:83]: Not needed. See my comment above

File: src/app/modules/document-more/document-bulk-changes/documents-choose-modification/documents-choose-modification.component.spec.ts [hunk 20]
```diff
@@ -17,10 +27,18 @@ describe('DocumentsChooseModificationComponent', () => {
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      declarations: [DocumentsChooseModificationComponent, TranslatePipeMock],
-      providers: [
-        { provide: TranslateService, useClass: TranslateServiceMock },
+      imports: [
+        HttpClientTestingModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-more/document-bulk-changes/documents-choose-modification/documents-choose-modification.component.spec.ts:32]: Use TranslateTestingModule

File: src/app/modules/document-more/document-bulk-changes/documents-choose-modification/documents-choose-modification.component.ts [hunk 21]
```diff
@@ -31,10 +35,12 @@ export class DocumentsChooseModificationComponent implements OnInit {
 
   constructor(
     private readonly translateService: TranslateService,
+    private userService: UserService,
     private readonly dataSharingService: DataSharingService
   ) {}
 
   ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-more/document-bulk-changes/documents-choose-modification/documents-choose-modification.component.ts:43]: Not needed. See my comment above

File: src/app/modules/document-more/document-bulk-changes/documents-choose/documents-choose.component.spec.ts [hunk 22]
```diff
@@ -26,10 +38,20 @@ describe('DocumentsChooseComponent', () => {
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       declarations: [DocumentsChooseComponent, TranslatePipeMock],
-      imports: [ReactiveFormsModule, HttpClientTestingModule],
+      imports: [
+        ReactiveFormsModule,
+        HttpClientTestingModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-more/document-bulk-changes/documents-choose/documents-choose.component.spec.ts:44]: Use TranslateTestingModule

File: src/app/modules/document-more/document-bulk-changes/documents-choose/documents-choose.component.ts [hunk 23]
```diff
@@ -28,13 +30,15 @@ export class DocumentsChooseComponent implements OnInit {
...
     private readonly translateService: TranslateService,
+    private userService: UserService,
     private readonly dataSharingService: DataSharingService,
     private readonly attachmentUploadService: AttachmentUploadService
   ) {
     this.results = this.dataSharingService.getSearchResults();
     this.onRowClick();
   }
   ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-more/document-bulk-changes/documents-choose/documents-choose.component.ts:41]: Not needed

File: src/app/modules/document-more/document-bulk-changes/documents-delete/documents-delete.component.spec.ts [hunk 24]
```diff
@@ -17,11 +26,18 @@ describe('DocumentsDeleteComponent', () => {
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      imports: [HttpClientTestingModule],
-      declarations: [DocumentsDeleteComponent, TranslatePipeMock],
-      providers: [
-        { provide: TranslateService, useClass: TranslateServiceMock },
+      imports: [
+        HttpClientTestingModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-more/document-bulk-changes/documents-delete/documents-delete.component.spec.ts:31]: Use TranslateTestingModule

File: src/app/modules/document-more/document-bulk-changes/documents-delete/documents-delete.component.ts [hunk 25]
```diff
@@ -21,10 +23,12 @@ export class DocumentsDeleteComponent implements OnInit {
 
   constructor(
     private readonly translateService: TranslateService,
+    private userService: UserService,
     private readonly attachmentUploadService: AttachmentUploadService
   ) {}
 
   ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-more/document-bulk-changes/documents-delete/documents-delete.component.ts:31]: Not needed

File: src/app/modules/document-more/document-more-routing.module.ts [hunk 26]
```diff
@@ -4,13 +4,11 @@ import { RouterModule, Routes } from '@angular/router';
 
 // Application imports
 import { DocumentBulkChangesComponent } from './document-bulk-changes/document-bulk-changes.component';
-import { CanActivateGuard } from 'src/app/shared/can-active-guard.service';
 
 const routes: Routes = [
   {
     path: 'bulkchanges',
```
Comment (SchettlerKoehler) [inline src/app/modules/document-more/document-more-routing.module.ts:10]: use addInitializeModuleGuard

File: src/app/modules/document-search/document-search-routing.module.ts [hunk 27]
```diff
@@ -4,13 +4,11 @@ import { RouterModule, Routes } from '@angular/router';
 
 // Application imports
 import { DocumentSearchComponent } from './document-search.component';
-import { CanActivateGuard } from 'src/app/shared/can-active-guard.service';
 
 const routes: Routes = [
   {
     path: '',
     component: DocumentSearchComponent,
-    canActivate: [CanActivateGuard],
```
Comment (SchettlerKoehler) [inline src/app/modules/document-search/document-search-routing.module.ts:13]: use addInitializeModuleGuard

File: src/app/modules/document-search/document-search.component.spec.ts [hunk 28]
```diff
@@ -54,6 +66,13 @@ describe('DocumentSearchComponent', () => {
         DropdownModule,
         AutoCompleteModule,
         SharedModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-search/document-search.component.spec.ts:69]: Use TranslateTestingModule

File: src/app/modules/document-search/document-search.component.spec.ts [hunk 29]
```diff
@@ -225,17 +244,23 @@ describe('DocumentSearchComponent', () => {
...
+    expect(component.items[1].label).toBe(
+      'DOCUMENT_SEARCH.FILTER.RECENTLY_UPDATED'
+    );
     expect(component.items[1].command).toBeDefined();
     spyOn(component, 'getFilteredRecentlyUpdated');
     component.items[1].command();
     expect(component.getFilteredRecentlyUpdated).toHaveBeenCalled();
-    expect(component.items[2].label).toBe(undefined);
+    expect(component.items[2].label).toBe(
+      'DOCUMENT_SEARCH.FILTER.CLEAR_FILTER'
```
Comment (SchettlerKoehler) [inline src/app/modules/document-search/document-search.component.spec.ts:262]: Why has this changed?

File: src/app/modules/document-search/document-search.component.ts [hunk 30]
```diff
@@ -23,14 +25,18 @@ import {
   GetDocumentByCriteriaRequestParams,
 } from 'src/app/generated';
 import { DataSharingService } from 'src/app/shared/data-sharing.service';
-import { convertToCSV } from 'src/app/utils';
+import { convertToCSV, getBestMatchLanguage } from 'src/app/utils';
```
Comment (SchettlerKoehler) [inline src/app/modules/document-search/document-search.component.ts:28]: use ExportService

File: src/app/modules/document-search/document-search.component.ts [hunk 31]
```diff
@@ -81,6 +88,7 @@ export class DocumentSearchComponent
   }
 
   public ngOnInit(): void {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-search/document-search.component.ts:91]: Not needed

File: src/app/modules/document-search/document-search.component.ts [hunk 32]
```diff
@@ -208,21 +218,27 @@ export class DocumentSearchComponent
   setFilterActions() {
     this.items = [
       {
-        label: this.translatedData['DOCUMENT_SEARCH.FILTER.CREATED_BY_ME'],
+        label: this.translateService?.instant(
```
Comment (SchettlerKoehler) [inline src/app/modules/document-search/document-search.component.ts:221]: Not not use translateService.instant(). See my comment above

File: src/app/modules/document-search/results/results.component.spec.ts [hunk 33]
```diff
@@ -70,13 +74,19 @@ describe('ResultsComponent', () => {
...
-      imports: [RouterTestingModule, HttpClientTestingModule],
-      declarations: [ResultsComponent, TranslatePipeMock],
-      providers: [
-        AttachmentUploadService,
-        { provide: TranslateService, useClass: TranslateServiceMock },
-        { provide: AUTH_SERVICE, useClass: IAuthMockService },
+      imports: [
+        RouterTestingModule,
+        HttpClientTestingModule,
+        TranslateModule.forRoot({
```
Comment (SchettlerKoehler) [inline src/app/modules/document-search/results/results.component.spec.ts:80]: Use TranslateTestingModule

File: src/app/modules/document-search/results/results.component.ts [hunk 34]
```diff
@@ -144,14 +150,15 @@ export class ResultsComponent implements OnInit {
...
-    @Inject(AUTH_SERVICE) private readonly authService: IAuthService,
+    private userService: UserService,
     private readonly router: Router,
     private readonly translateService: TranslateService,
     private readonly activeRoute: ActivatedRoute,
     private readonly attachmentUploadService: AttachmentUploadService
   ) {}
 
   ngOnInit() {
+    this.translateService.use(getBestMatchLanguage(this.userService));
```
Comment (SchettlerKoehler) [inline src/app/modules/document-search/results/results.component.ts:161]: Not needed

File: src/test.ts [hunk 35]
```diff
@@ -7,16 +7,16 @@ import {
...
+// declare const require: {
+//   context(
+//     path: string,
+//     deep?: boolean,
+//     filter?: RegExp
+//   ): {
+//     keys(): string[];
+//     <T>(id: string): T;
+//   };
+// };
```
Comment (SchettlerKoehler) [inline src/test.ts:19]: Remove commented code

File: src/test.ts [hunk 36]
```diff
@@ -27,6 +27,6 @@ getTestBed().initTestEnvironment(
   }
 );
 // Then we find all the tests.
-const context = require.context('./', true, /\.spec\.ts$/);
-// And load the modules.
-context.keys().forEach(context);
+// const context = require.context('./', true, /\.spec\.ts$/);
+// // And load the modules.
+// context.keys().forEach(context);
```
Comment (SchettlerKoehler) [inline src/test.ts:32]: Same here

**Review by SchettlerKoehler — CHANGES_REQUESTED** [review]

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (insane-04) [reply src/app/app-routing.module.ts:8]: Added

File: src/app/utils.ts (hunk 3 above)
Comment (insane-04) [reply src/app/utils.ts]: removed!

File: src/test.ts (hunk 35 above)
Comment (insane-04) [reply src/test.ts]: removed

File: src/app/modules/document-detail/document-create/document-create.component.spec.ts (hunk 4 above)
Comment (insane-04) [reply src/app/modules/document-detail/document-create/document-create.component.spec.ts]: Not using now

File: src/app/modules/document-detail/document-detail-routing.module.ts (hunk 6 above)
Comment (insane-04) [reply src/app/modules/document-detail/document-detail-routing.module.ts:13]: Added addInitializeModuleGuard

File: src/app/modules/document-search/results/results.component.ts (hunk 34 above)
Comment (insane-04) [reply src/app/modules/document-search/results/results.component.ts]: not using !! after adding addInitializerModule the translation is working proper.

File: src/app/modules/document-detail/document-edit/document-edit.component.spec.ts (hunk 9 above)
Comment (insane-04) [reply src/app/modules/document-detail/document-edit/document-edit.component.spec.ts]: now we are not using TranslateModule

File: src/app/modules/document-more/document-bulk-changes/documents-delete/documents-delete.component.spec.ts (hunk 24 above)
Comment (insane-04) [reply src/app/modules/document-more/document-bulk-changes/documents-delete/documents-delete.component.spec.ts]: TranslateModule is not using now

File: src/app/modules/document-search/document-search-routing.module.ts (hunk 27 above)
Comment (insane-04) [reply src/app/modules/document-search/document-search-routing.module.ts:13]: Added

File: src/app/modules/document-search/document-search.component.spec.ts (hunk 29 above)
Comment (insane-04) [reply src/app/modules/document-search/document-search.component.spec.ts]: I reverted the changes, Previously! the testcases are failing due to translation issue.

File: src/app/modules/document-detail/document-quick-upload/document-quick-upload-form/document-quick-upload-form.component.ts (hunk 15 above)
Comment (insane-04) [reply src/app/modules/document-detail/document-quick-upload/document-quick-upload-form/document-quick-upload-form.component.ts]: removed

File: src/app/modules/document-search/document-search.component.ts [hunk 37]
```diff
@@ -336,9 +339,10 @@ export class DocumentSearchComponent
         show: 'asOverflow',
         actionCallback: () => {
           if (this.results.length == 0) {
-            this.messageService.add({
-              severity: 'success',
-              summary: this.translatedData['GENERAL.NO_RECORDS_TO_EXPORT'],
+            this.portalMessageService.success({
+              summaryKey: this.translateService.instant(
```
Comment (SchettlerKoehler) [inline src/app/modules/document-search/document-search.component.ts:343]: Same here

File: src/app/modules/document-search/document-search.component.ts [hunk 38]
```diff
@@ -362,9 +366,10 @@ export class DocumentSearchComponent
         show: 'asOverflow',
         actionCallback: () => {
           if (this.results.length == 0) {
-            this.messageService.add({
-              severity: 'success',
-              summary: this.translatedData['GENERAL.NO_RECORDS_FOR_CHANGES'],
+            this.portalMessageService.success({
+              summaryKey: this.translateService.instant(
```
Comment (SchettlerKoehler) [inline src/app/modules/document-search/document-search.component.ts:370]: Same here

File: src/app/utils.ts [hunk 39]
```diff
@@ -35,7 +40,7 @@ export function createTranslateLoader(http: HttpClient, mfeInfo: MfeInfo) {
 }
 
 export const basePathProvider = (mfeInfo: MfeInfo) => {
-  console.log(`Base path provider ${mfeInfo?.remoteBaseUrl}`);
+  console.log(`Base path provider ${environment?.API_BASE_PATH}`);
```
Comment (SchettlerKoehler) [inline src/app/utils.ts:43]: This is not the same. Why the change?

**Review by SchettlerKoehler — CHANGES_REQUESTED** [review]

File: src/app/modules/document-detail/document-quick-upload/document-quick-upload.component.ts [hunk 40]
```diff
@@ -358,16 +358,14 @@ export class DocumentQuickUploadComponent implements OnInit {
...
-              severity: 'success',
-              summary: `${successFiles} ${this.translatedData['DOCUMENT_DETAIL.MULTIPLE_ATTACHMENTS.UPLOAD_SUCCESS']}`,
+            this.portalMessageService.success({
+              summaryKey: `${successFiles} ${this.translatedData['DOCUMENT_DETAIL.MULTIPLE_ATTACHMENTS.UPLOAD_SUCCESS']}`,
             });
           }
           if (failedFiles > 0) {
-            this.messageService.add({
-              severity: 'error',
+            this.portalMessageService.error({
```
[the remark this answers is not in the document]
Comment (insane-04) [reply src/app/modules/document-detail/document-quick-upload/document-quick-upload.component.ts:367]: resolved

File: src/app/modules/document-search/document-search.component.ts (hunk 38 above)
Comment (insane-04) [reply src/app/modules/document-search/document-search.component.ts]: used only summaryKey
